src/wp-includes/class-wp-query.php

```diff
@@ -3066,17 +3073,38 @@ public function get_posts() {
 		";
 
 		$this->request = $old_request;
+		$this->count_request = "
+			SELECT COUNT(DISTINCT {$count_field})
```

Comment:
shouldn't `DISTINCT` be `$distinct`. Adding when its not needed results in more work for the DB.

Reply:
As you're counting "found rows" here, you really don't want it to be distinct (as distinct rows <= actual rows). So technically `DISTINCT` should be omitted.

`{$count_field}` could be left as a `*` literal like the MySQL documents suggest, so if a `$where` clause exist, to choose from whatever available index suites the query best.

Look at the `explain` plan for when there's a WHERE clause that can use an index. Might not make a difference. Also see explain plain when a JOIN is added.

Reply:
Regarding DISTINCT, I'll check this again to be 100% sure but I think this needs to remain in place because some queries can result in duplicate rows being returned. These tests cover that scenario:

* `test_found_posts_are_correct_for_OR_meta_queries()`
* `test_found_posts_are_correct_for_group_by_queries()`

Reply:
I checked this again and `SELECT COUNT(DISTINCT {$count_field})` is indeed needed for the scenarios covered by the two tests above. If there was a way to only add the `DISTINCT` clause when we knew we needed it that would be great, but I'm not too confident about that.